File: app/src/main/java/com/moufee/purduemenus/ui/menu/MenuItemHolder.java
package com.moufee.purduemenus.ui.menu;

import com.moufee.purduemenus.databinding.FragmentMenuitemBinding;
import com.moufee.purduemenus.menus.MenuItem;
import com.moufee.purduemenus.menus.OnToggleFavoriteListener;

import androidx.recyclerview.widget.RecyclerView;

/**
 * Created by Ben on 14/07/2017.
 * RecyclerView ViewHolder for a menu item
 */

public class MenuItemHolder extends RecyclerView.ViewHolder {
    public FragmentMenuitemBinding menuitemBinding;


    public MenuItemHolder(FragmentMenuitemBinding binding) {
        super(binding.getRoot());
        menuitemBinding = binding;
    }

    public void bind(MenuItem item, OnToggleFavoriteListener listener, boolean isFavorite) {
        menuitemBinding.setMenuItem(item);
        menuitemBinding.setIsFavorite(isFavorite);
        menuitemBinding.setListener(listener);
        menuitemBinding.executePendingBindings();
    }

}

File: app/src/main/java/com/moufee/purduemenus/ui/menu/MenuItemHolder.kt
package com.moufee.purduemenus.ui.menu

import androidx.recyclerview.widget.RecyclerView
import com.moufee.purduemenus.databinding.FragmentMenuitemBinding
import com.moufee.purduemenus.repository.data.menus.MenuItem

/**
 * Created by Ben on 14/07/2017.
 * RecyclerView ViewHolder for a menu item
 */
class MenuItemHolder(private val menuItemBinding: FragmentMenuitemBinding) : RecyclerView.ViewHolder(menuItemBinding.root) {
    fun bind(item: MenuItem?, listener: OnToggleFavoriteListener?, isFavorite: Boolean) {
        menuItemBinding.menuItem = item
        menuItemBinding.isFavorite = isFavorite
        menuItemBinding.listener = listener
        menuItemBinding.executePendingBindings()
    }
}